api/lib/block-listeners/cosmos-node-subscription.js

```diff
@@ -51,6 +53,25 @@ class CosmosNodeSubscription {
     }, PROPOSAL_POLLING_INTERVAL)
   }
 
+  async updateValidators() {
+    const cosmosAPI = new this.CosmosApiClass(this.network, this.store)
+    let validators = await cosmosAPI.getAllValidators(this.height)
+    validators = await this.addPopularityToValidators(
+      validators,
+      this.network.id
+    )
+    const validatorMap = await this.getValidatorMap(validators)
+    this.updateDBValidatorProfiles(validators)
+    this.store.update({
```

Comment:
you could run

```
 validators = await this.addPopularityToValidators(
      validators,
      this.network.id
    )
```

in the store for every update. this way it scales across polkadot

you would leave the other update in place

Reply:
Thank you for this observation! I just applied it and it goes even faster :star_struck: 

I also think the loop think is a great idea. Doesn't make sense to me updating validators every block.

Let's merge and enjoy :cocktail:

Reply:
> I also think the loop think is a great idea. Doesn't make sense to me updating db validators every block.

still we want the live validators even if we don't enrich them from db every block

Reply:
Yes, I understand your concern. I think what I'm going to do in the end is:

- keep loop in the listener only for updateDBValidators

- move the fixed `addPopularity`thing back again to the resolver

- everything else stays the same

End of the problem

Reply:
Can we move the loop for updating the db validators also into the store so it is available on all networks the same?

Reply:
Ouch I hadn't seen this one. OK, let's do it

Reply:
It happens the same. Slows everything down... :shrug:  So strange Seems store is quite delicate

Reply:
:thinking:  :sweat:  I think I never managed to fix this.. noooo